tests/jballerina-unit-test/src/test/resources/test-src/statements/matchstmt/varbindingpatternmatchpattern/list_binding_pattern.bal

```diff
@@ -501,6 +501,29 @@ function testListBindingPattern22() {
     assertEquals((), listBindingPattern22(1));
 }
 
+function listBindingPattern23(any a) returns string {
+    match a {
+        var [] => {
+            return "match1";
+        }
+        var [s, [...i]] => {
+            return "match2";
+        }
+        var [...i] => {
+            return "match3";
+        }
+        _ => {
+            return "No match";
+        }
+    }
+}
```

Comment:
@TharushiJay, can you check whether this same example works for the `list-match-pattern`?

Reply:
Seems like it runs into the same error as well. I'll fix it in the same PR.